anonlink/similarities/hamming.py

```diff
@@ -3,120 +3,112 @@
 import heapq
 import itertools
 import operator
-from typing import (cast, DefaultDict, Dict, Iterable,
+from array import array
+from typing import (cast, Counter, DefaultDict, Dict, Iterable,
                     List, Optional, Sequence, Tuple)
 
 import numpy as np
+from bitarray import bitarray
 
-from anonlink.typechecking import Record
+from anonlink.typechecking import Dataset, FloatArrayType, IntArrayType
 
 
-def _hamming_sim(clk1: Record, clk2: Record) -> float:
+_Similarity = Tuple[float, int, int]
+
+
+def _hamming_sim(clk1: bitarray, clk2: bitarray) -> float:
     assert len(clk1) == len(clk2)
     assert len(clk1) != 0
-    return sum(map(operator.eq, clk1, clk2)) / len(clk1)
+    return 1 - (clk1 ^ clk2).count() / len(clk1)
```

Comment:
sneaky

Reply:
This _kind of_ made sense to put there. I changed the type annotations to support only `bitarray`s instead of the more general `Sequence[bool]`. I thought the similarity code should reflect that.